src/main/java/com/couchbase/lite/util/Utils.java

```diff
@@ -124,11 +116,8 @@ public static boolean isTransientError(Response response) {
     }
 
     public static boolean isTransientError(int code) {
-        if (code == 500 || code == 502 || code == 503 || code == 504 ||
-                code == Status.REQUEST_TIMEOUT) {
-            return true;
-        }
-        return false;
+        // Note: Status.NOT_ACCEPTABLE = 406, Status.REQUEST_TIMEOUT = 408
+        return code == 406 || code == 408 || code == 500 || code == 502 || code == 503 || code == 504;
```

Comment:
In general a transient error means that it's temporary and recoverable. For 406 with CouchDB, is it recoverable?